cibuildwheel/options.py

```diff
@@ -673,10 +673,17 @@ def globals(self) -> GlobalOptions:
         )
 
     def _check_pinned_image(self, value: str, pinned_images: Mapping[str, str]) -> None:
-        if (
-            value in {"manylinux1", "manylinux2010", "manylinux_2_24", "musllinux_1_1"}
-            and value not in self._image_warnings
-        ):
+        error_set = {"manylinux1", "manylinux2010", "manylinux_2_24", "musllinux_1_1"}
+        warning_set = error_set  # shall include error_set
+
+        if value in warning_set and value not in self._image_warnings:
+            if value in error_set:
+                msg = (
+                    f"cibuildwheel 3.x does not support the image {value!r}. Either upgrade to a "
+                    "supported image or continue using the image by pinning it directly with"
+                    " its full OCI registry '<name>{:<tag>|@<digest>}'."
+                )
+                raise errors.DeprecationError(msg)
```

Comment:
We can simplify a bit, I think, decouple errors and warnings...

```suggestion
        error_set = {"manylinux1", "manylinux2010", "manylinux_2_24", "musllinux_1_1"}
        warning_set = set()

        if value in error_set:
            msg = (
                f"cibuildwheel 3.x does not support the image {value!r}. Either upgrade to a "
                "supported image or continue using the image by pinning it directly with"
                " its full OCI registry '<name>{:<tag>|@<digest>}'."
            )
            raise errors.DeprecationError(msg)

        if value in warning_set and value not in self._image_warnings:
```